classes/shop/Shop.php

```diff
@@ -464,13 +465,36 @@ public function getAddress()
     }
 
     /**
-     * Get shop theme name
-     *
-     * @return string
+     * Set shop theme details from Json data
      */
+    public function setTheme()
+    {
+        $dir = _PS_ALL_THEMES_DIR_.$this->theme_name;
+        $theme_data = Yaml::parse(file_get_contents(
```

Comment:
Good idea. I'm not sure anyone wants to use XML but the Config component will let us add constraints to this file.